sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaPlugin.java

```diff
@@ -47,10 +48,14 @@ public void define(Context context) {
       builder.addAll(SurefireExtensions.getExtensions());
       builder.addAll(JaCoCoExtensions.getExtensions(sonarQubeVersion));
       builder.add(JavaSonarWayProfile.class);
+      builder.addAll(JavaClasspathProperties.getProperties());
+      builder.add(JavaClasspath.class);
+    } else {
+      builder.addAll(JavaClasspathProperties.getProperties());
```

Comment:
Order is not important, pico collect all classes, and then instantiate them depending on the dependencies.